examples/camera_poc.cc

```diff
@@ -0,0 +1,189 @@
+#include <SDL2_mixer/SDL_mixer.h>
+#include <stdio.h>
+
+#include <vector>
+
+#include "core/include/sdl_handler.h"
+
+const int SIZE = 40;
+const int NUM_ROW = 15; // 800*600 => 600 / SIZE = 15
+const int NUM_COL = 50; // 800*600 => 800 / SIZE = 50
+
+class Camera {
+   public:
+    SDL_Rect viewport;
+    Camera(int width, int height) {
+        viewport.x = 0;
+        viewport.y = 0;
+        viewport.w = width;
+        viewport.h = height;
+    }
+
+    // update the camera position based on the player position
+    void update(int playerX, int playerY, int screenWidth, int screenHeight) {
+        viewport.x = playerX + SIZE / 2 - screenWidth / 2;
+        viewport.y = playerY + SIZE / 2 - screenHeight / 2;
+
+        // avoid camera to go out of the screen
+        if (viewport.x < 0)
+            viewport.x = 0;
+        if (viewport.y < 0)
+            viewport.y = 0;
+
+        // assume the screen is 800x600
+        if (viewport.x + viewport.w > NUM_COL * SIZE)
+            viewport.x = NUM_COL * SIZE - viewport.w;
+        if (viewport.y + viewport.h > NUM_ROW * SIZE)
+            viewport.y = NUM_ROW * SIZE - viewport.h;
+    }
+
+    // apply the camera to a rect
+    SDL_Rect apply(SDL_Rect rect) const {
```

Comment:
nit: No need to copy this obj.
```suggestion
    SDL_Rect apply(SDL_Rect& rect) const {
```

Reply:
revised